tests/citycoin-vote-v1.test.ts

```diff
@@ -0,0 +1,874 @@
+        receipt.result.expectErr().expectUint(VoteModel.ErrCode.ERR_UNAUTHORIZED);
+      });
+      it("fails with ERR_UNAUTHORIZED if the start height is before the current height", () => {
+        // arrange
+        const wallet = accounts.get("deployer")!;
+        const startHeight = 8500;
+        const endHeight = 10600;
+        // act
+        chain.mineEmptyBlockUntil(startHeight + 1);
+        chain.mineBlock([
+          vote.initializeContract(startHeight, endHeight, wallet)
+        ]);
+        const receipt = chain.mineBlock([
+          vote.initializeContract(startHeight, endHeight, wallet)
+        ]).receipts[0];
+        // assert
+        receipt.result.expectErr().expectUint(VoteModel.ErrCode.ERR_UNAUTHORIZED);
+      });
+      it("fails with ERR_UNAUTHORIZED if the end height is before the start height", () => {
+        // arrange
+        const wallet = accounts.get("deployer")!;
+        const startHeight = 8500;
+        const endHeight = 8400;
+        // act
+        chain.mineBlock([
+          vote.initializeContract(startHeight, endHeight, wallet)
+        ]);
+        const receipt = chain.mineBlock([
+          vote.initializeContract(startHeight, endHeight, wallet)
+        ]).receipts[0];
+        // assert
+        receipt.result.expectErr().expectUint(VoteModel.ErrCode.ERR_UNAUTHORIZED);
+      });
+      it("succeeds and updates the start and end block height variables", () => {
+        // arrange
+        const wallet = accounts.get("deployer")!;
+        const startHeight = 8500;
+        const endHeight = 10600;
+        // act
+        const receipt = chain.mineBlock([
+          vote.initializeContract(startHeight, endHeight, wallet)
+        ]).receipts[0];
+        // assert
+        receipt.result.expectOk();
+      });
+    })
+    describe("vote-on-proposal()", () => {
+      it("fails with ERR_PROPOSAL_NOT_ACTIVE when called before proposal is active", () => {
+        // arrange
+        const wallet = accounts.get("wallet_1")!;
+        const deployer = accounts.get("deployer")!;
+        const startHeight = 8500;
+        const endHeight = 10600;
+        chain.mineBlock([
+          vote.initializeContract(startHeight, endHeight, deployer)
+        ]);
+        // act
+        const receipt = chain.mineBlock([
+          vote.voteOnProposal(true, wallet)
+        ]).receipts[0];
+        // assert
+        receipt.result.expectErr().expectUint(VoteModel.ErrCode.ERR_PROPOSAL_NOT_ACTIVE);
+      });
+      it("fails with ERR_PROPOSAL_NOT_ACTIVE when called after proposal is active", () => {
+        // arrange
+        const wallet = accounts.get("wallet_1")!;
+        const deployer = accounts.get("deployer")!;
+        const startHeight = 8500;
+        const endHeight = 10600;
+        chain.mineBlock([
+          vote.initializeContract(startHeight, endHeight, deployer)
+        ]);
+        chain.mineEmptyBlock(VoteModel.VOTE_START_BLOCK + VoteModel.VOTE_END_BLOCK + 1);
```

Comment:
```suggestion
        chain.mineEmptyBlockUntil(VoteModel.VOTE_END_BLOCK + 1);
```
Moving chain tip to `VOTE_END_BLOCK + 1` (10601) feels a bit more appropriate than `VOTE_START_BLOCK + VOTE_END_BLOCK + 1` (19101).

Reply:
Yes, good catch, I think my over-complicated intent was actually `VOTE_START_BLOCK + (VOTE_END_BLOCK - VOTE_START_BLOCK) + 1`